src/write/macho.rs

```diff
@@ -63,9 +63,76 @@ impl Object {
                 &b"__bss"[..],
                 SectionKind::UninitializedData,
             ),
+            StandardSection::Tls => (&b"__DATA"[..], &b"__thread_data"[..], SectionKind::Tls),
+            StandardSection::UninitializedTls => (
+                &b"__DATA"[..],
+                &b"__thread_bss"[..],
+                SectionKind::UninitializedTls,
+            ),
+            StandardSection::TlsVariables => (
+                &b"__DATA"[..],
+                &b"__thread_var"[..],
+                SectionKind::TlsVariables,
+            ),
+        }
+    }
+
+    fn macho_tlv_bootstrap(&mut self) -> SymbolId {
+        match self.tlv_bootstrap {
+            Some(id) => id,
+            None => {
+                let id = self.add_symbol(Symbol {
+                    name: b"_tlv_bootstrap".to_vec(),
+                    value: 0,
+                    size: 0,
+                    kind: SymbolKind::Text,
+                    scope: SymbolScope::Dynamic,
+                    weak: false,
+                    section: None,
+                });
+                self.tlv_bootstrap = Some(id);
+                id
+            }
         }
     }
 
+    pub(crate) fn macho_add_thread_var(&mut self, symbol: SymbolId) {
```

Comment:
When codegening a tls access for mach-O you need to use the symbol in the `__thread_vars` section as relocation target. The actual data needs to be stored at the symbol in the `__thread_data` section instead. LLVM adds a postfix of `$tlv$init` to that one.

Reply:
Should be a bit closer to correct now.